### PR TITLE
Support for WIX

### PR DESCRIPTION
**Pull request by jakebrown58**
* Can uninstall WIX installations.
* Can parse quotes properly for WIX installers.
* Fixes an encryption regression in Master.exe, where it didn't work if
you were opting out of encryption.

### CODE REVIEW

File: src/AutoDeploy/Master/Program.cs [hunk 1]
```diff
@@ -231,31 +237,33 @@ public static int DoIt(Logger logger, Options o)
...
 
                 string keyArg = "-e " + base64EncryptionKey;
                 string vectorArg = "-v " + base64IV;
-                string encryptionArgs = keyArg + " " + vectorArg;
-
-                var genericInstallWithKeys = new List<string>();
+                encryptionArgs = keyArg + " " + vectorArg;
+            }
+            catch
+            {
```
Comment (adampaulberg) [inline src/AutoDeploy/Master/Program.cs]: do you want to do something here, at least even log it?

File: src/AutoDeploy/Master/App/ConfigurationHelper.cs [hunk 2]
```diff
@@ -43,11 +43,11 @@ public static string FillInParametersForCommand(string command, KeyValueConfigDi
 
                 var parameterValue = parameter;
                 lookupKeys.TryGetValueForKey(appLookupKey, parameter, out parameterValue);
-                Console.WriteLine("             replaced key: " + parameter + " with value: " + parameterValue);
+                //Console.WriteLine("             replaced key: " + parameter + " with value: " + parameterValue);
```
Comment (adampaulberg) [inline src/AutoDeploy/Master/App/ConfigurationHelper.cs]: maybe remove?

File: src/AutoDeploy/Master/App/ConfigurationHelper.cs [hunk 3]
```diff
@@ -43,11 +43,11 @@ public static string FillInParametersForCommand(string command, KeyValueConfigDi
...
                 realCommand += key + parameterValue;
             }
 
-            Console.WriteLine("          reading command as: " + realCommand);
+            //Console.WriteLine("          reading command as: " + realCommand);
```
Comment (adampaulberg) [inline src/AutoDeploy/Master/App/ConfigurationHelper.cs]: maybe remove?

File: src/AutoDeploy/UninstallerHelper/Program.cs [hunk 4]
```diff
@@ -25,14 +25,13 @@ static int Main(string[] args)
                 var allUninstallStrings = new List<string>();
 
                 string matchBy = String.Empty;
-                if(args.Length == 0 )
+                if (args.Length == 0)
```
Comment (adampaulberg) [inline src/AutoDeploy/UninstallerHelper/Program.cs]: I could be reading this wrong, but won't args[0] throw an exception if length is 0?

File: src/AutoDeploy/Master/Program.cs (hunk 1 above)
Comment (jakebrown58) [inline src/AutoDeploy/Master/Program.cs]: good eye.  this is a little wonky, but I'm happy with this change here.   it's a dark feature, and i'm doing the minimum necessary to make the base-case (no encryption) continue to work.

when we turn on encryption for real, we'll be improving the 'try' area, and probably just bail out on a 'no decryption provided' kind of a warning.

File: src/AutoDeploy/Master/App/ConfigurationHelper.cs (hunk 2 above)
Comment (jakebrown58) [inline src/AutoDeploy/Master/App/ConfigurationHelper.cs]: ok.

File: src/AutoDeploy/Master/App/ConfigurationHelper.cs (hunk 3 above)
Comment (jakebrown58) [inline src/AutoDeploy/Master/App/ConfigurationHelper.cs]: ok.

File: src/AutoDeploy/UninstallerHelper/Program.cs (hunk 4 above)
Comment (jakebrown58) [inline src/AutoDeploy/UninstallerHelper/Program.cs]: yeah, that block of code is junk.   ...but this code only ever gets called programatically, and we don't have issues with un-installations, so I 'm not inclined to adjust it right now.

Comment (jakebrown58) [inline src/AutoDeploy/UninstallerHelper/Program.cs]: that whole area is just a hack for the Ringtail Configurator which installed with a different pattern name than everything else.... it's dead code, but harmless.

Probably the right way to fix it is if the arg length passed in is 0, to show 'help' or 'usage'.

**Pull request merged**